### PR TITLE
Update tutorials of both "Building a custom collector" and "Building a Trace Receiver"

### PR DESCRIPTION
**Pull request by brightzheng100**
Re issue: https://github.com/open-telemetry/opentelemetry.io/issues/3488

This PR has two tutorials / files updated.

**In "Building a custom collector":**
1. Added the folder structure to help readers understand where the files should be properly located.
2. Added the further readings as the additional resources.

**In "Building a Trace Receiver":**
1. Streamlined & simplified the development process;
3. Applied the Go module / workspaces practices with a better folder structure;
4. Fixed many broken links due to a missing `v` in the path. For example: `/blob/v{{%
param collectorVersion %}}/pdata/`;
5. Fixed other minor wording issues / typos.

### CODE REVIEW

**Comment by svrnm** [comment]
Hey @brightzheng100, thanks for taking the effort to improve those 2 documents, that's highly appreciated. I will try to provide you with a feedback soon.

@open-telemetry/collector-approvers please take a look as well!

File: content/en/docs/collector/trace-receiver.md
Comment (svrnm) [inline content/en/docs/collector/trace-receiver.md:940]:
```suggestion
Till now, the skeleton of the receiver has been fully implemented.
```

File: content/en/docs/collector/trace-receiver.md
Comment (svrnm) [inline content/en/docs/collector/trace-receiver.md:1100]:
```suggestion
you normally debug a Go project. Here is a simple `launch.json` file for
```

File: content/en/docs/collector/trace-receiver.md
Comment (svrnm) [inline content/en/docs/collector/trace-receiver.md:1924]:
```suggestion
- Added new functions `NewTraceID()` and `NewSpanID()` to generate trace ID and
```

File: content/en/docs/collector/trace-receiver.md
Comment (svrnm) [inline content/en/docs/collector/trace-receiver.md:1925]:
```suggestion
  span ID respectively.
```

**Review by svrnm — APPROVED** [review]
Overall it looks good to me. Please add `telemetrygen` to the cSpell:ignore list in line 5

@open-telemetry/collector-approvers PTAL!

File: content/en/docs/collector/custom-collector.md
Comment (chalin) [inline content/en/docs/collector/custom-collector.md:181]:
```suggestion
Further reading:
```

**Review by chalin — APPROVED** [review]

**Comment by cartermp** [comment]
@open-telemetry/collector-approvers please take a look!

**Comment by svrnm** [comment]
I recently tried out the tracer receiver tutorial by using the updates from this PR and it worked just fine 👍

File: content/en/docs/collector/trace-receiver.md
Comment (jpkrohling) [inline content/en/docs/collector/trace-receiver.md:89]:
```suggestion
Once the container is up and running, you can access Jaeger UI by:
```

File: content/en/docs/collector/trace-receiver.md
Comment (jpkrohling) [inline content/en/docs/collector/trace-receiver.md:100]: Why is `batch` included here? I would keep only the things that are relevant for the tutorial. If the idea is to have a recommended configuration in place, then it's missing the memory limiter, pprof, and possibly others.

File: content/en/docs/collector/trace-receiver.md
Comment (jpkrohling) [inline content/en/docs/collector/trace-receiver.md:134]: We have docs here somewhere on how to use TLS. Perhaps we can reuse that instead of showing how to do things insecurely?

File: content/en/docs/collector/trace-receiver.md
Comment (jpkrohling) [inline content/en/docs/collector/trace-receiver.md:166]: Should this URL be parameterized as well?

**Review by jpkrohling — APPROVED** [review]
Looks sane, but I didn't try this. Given that @svrnm did, I'm approving this, although I'd love to see the couple of changes I suggested.

File: content/en/docs/collector/trace-receiver.md
Comment (brightzheng100) [reply content/en/docs/collector/trace-receiver.md:100]: As the "Building a custom collector" doc is kind of the starting point, I respect the original doc [here](https://opentelemetry.io/docs/collector/custom-collector/), where the `batch` processor is included.

That's why I wanted to make them aligned but if you wanted to remove such an "optional" component, I can remove it and may send another PR to update that "Building a custom collector" doc as well.

File: content/en/docs/collector/trace-receiver.md
Comment (brightzheng100) [reply content/en/docs/collector/trace-receiver.md:166]: That's a good idea but this URL might be stable enough to last for a while I believe.

File: content/en/docs/collector/trace-receiver.md
Comment (brightzheng100) [reply content/en/docs/collector/trace-receiver.md:134]: Yes, the "Configuration" doc [here](https://opentelemetry.io/docs/collector/configuration/#setting-up-certificates) does mention how. Maybe we can add a reference link to that section while keeping the tutorial focused?

**Comment by svrnm** [comment]
@brightzheng100 can you fix the merge conflicts?

File: content/en/docs/collector/trace-receiver.md
Comment (svrnm) [reply content/en/docs/collector/trace-receiver.md:166]: can this conversation be resolved?

File: content/en/docs/collector/trace-receiver.md
Comment (svrnm) [reply content/en/docs/collector/trace-receiver.md:134]: can this conversation be resolved?

**Comment by brightzheng100** [comment]
There are quite a lot of changes after my PR as it's been a while.

But I've tried to merge and test so you may have a final review and let me know if further changes are needed.

**Comment by cartermp** [comment]
@brightzheng100 I think if you run `npm run fix` it should clear up the rest. The content looks great to me, though. When those issues get fixed I'll merge it.

**Comment by brightzheng100** [comment]
Done!

I was seeing exceptions when running `npm run fix`, as attached below, so I stepped back by using only `npm run fix:format` and `npm run serve` to verify the contents before submitting.

But actually if I were to look into those exceptions, I should know how to fix those issues one by one based on the exception messages.

Not sure whether this is the right way like what I did (by looking into the exceptions) for contributors to review before `git commit` and `git push`, or there are some bugs in `lint-md`.

```
...
[10:52:14] Starting 'lint-md'...
/Users/brightzheng/workspaces/OSS/OpenTelemetry/opentelemetry.io-mine/content/en/docs/collector/custom-collector.md:162: MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
/Users/brightzheng/workspaces/OSS/OpenTelemetry/opentelemetry.io-mine/content/en/docs/collector/building/receiver.md:201: MD034/no-bare-urls Bare URL used [Context: "https://github.com/rquedas/ote..."]
/Users/brightzheng/workspaces/OSS/OpenTelemetry/opentelemetry.io-mine/content/en/docs/collector/building/receiver.md:60: MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
/Users/brightzheng/workspaces/OSS/OpenTelemetry/opentelemetry.io-mine/content/en/docs/collector/building/receiver.md:1121: MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
[10:52:16] 'lint-md' errored after 1.83 s
[10:52:16] Error: Processed 364 files, 2 had issues.
    at Transform.<anonymous> (/Users/brightzheng/workspaces/OSS/OpenTelemetry/opentelemetry.io-mine/gulp-src/lint-md.js:152:15)
    at Transform.emit (node:events:526:35)
    at Transform.emit (node:domain:551:15)
    at endReadableNT (/Users/brightzheng/workspaces/OSS/OpenTelemetry/opentelemetry.io-mine/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:1003:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

**Comment by cartermp** [comment]
Huh, interesting. Maybe it's a bug in the linting tool we use.

**Review by cartermp — APPROVED** [review]

**Pull request merged**
